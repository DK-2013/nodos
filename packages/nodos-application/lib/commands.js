import _ from 'lodash';
import jest from 'jest';
import repl from 'repl';
import columnify from 'columnify';
import log from './logger';

export const testCommand = ({ container }) => ({
  command: 'test [file]',
  describe: 'run tests',
  builder: {},
  handler: (argv) => {
    // FIXME: not working
    process.env.NODOS_ENV = 'test';
    // const rest = argv._.slice(1);
    const options = [];
    if (argv.file) {
      options.push('-f', argv.file);
    } else {
      options.push('--testPathPattern', '/tests/');
    }
    const jestItem = _.get(container, 'jest', jest);
    log(options);
    jestItem.run(options);
  },
});

export const consoleCommand = ({ app, container }) => ({
  command: 'console',
  describe: 'run console',
  builder: {},
  handler: async () => {
    const replItem = _.get(container, 'repl', repl);
    const replServer = replItem.start({
      prompt: '> ',
    });
    replServer.context.app = app;
  },
});

export const serverCommand = ({ app }) => ({
  command: 'server',
  describe: 'run server',
  builder: yargs => yargs
    .default('h', '127.0.0.1')
    .alias('h', 'host')
    .default('p', 3000)
    .alias('p', 'port'),
  handler: async (argv) => {
    app.listen(argv.port, argv.host, (err, address) => {
      if (err) {
        console.error(err);
        process.exit(1);
      }
      console.log(address);
    });
  },
});

export const routesCommand = ({ app, container }) => ({
  command: 'routes',
  describe: 'display routes',
  handler: async () => {
    const print = _.get(container, 'print', console.log);
    const { routes } = app.router;
    if (_.isEmpty(routes)) {
      print(
        'You don\'t have any routes defined!\n\n'
        + 'Please add some routes in config/routes.yml.',
      );
    } else {
      const formattedRoutes = columnify(
        routes,
        {
          columns: ['method', 'url', 'pipeline'],
          config: {
            method: {
              headingTransform: () => 'Verb',
              dataTransform: _.toUpper,
            },
            url: {
              headingTransform: () => 'URI Pattern',
            },
            pipeline: {
              headingTransform: () => 'Pipeline',
            },
          },
        },
      );
      print(formattedRoutes);
    }
  },
});
