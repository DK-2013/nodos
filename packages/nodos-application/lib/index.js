import BaseApplication from './BaseApplication';

export default BaseApplication;
