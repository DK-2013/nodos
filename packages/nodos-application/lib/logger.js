import debug from 'debug';

export default debug('nodos:application');
