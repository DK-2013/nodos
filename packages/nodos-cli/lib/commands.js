const _ = require('lodash');
// const jest = require('jest');
const repl = require('repl');
const columnify = require('columnify');

// const testCommand = ({ container }) => ({
//   command: 'test [file]',
//   describe: 'run tests',
//   builder: {},
//   handler: (argv) => {
//     // FIXME: not working
//     process.env.NODOS_ENV = 'test';
//     // const rest = argv._.slice(1);
//     const options = [];
//     if (argv.file) {
//       options.push('-f', argv.file);
//     } else {
//       options.push('--testPathPattern', '/tests/');
//     }
//     const jestItem = _.get(container, 'jest', jest);
//     log(options);
//     jestItem.run(options);
//   },
// });

const buildConsoleCommand = ({ app, container }) => ({
  command: 'console',
  describe: 'run console',
  builder: {},
  handler: async () => {
    const actualRepl = _.get(container, 'repl', repl);
    const replServer = actualRepl.start({
      prompt: '> ',
    });
    replServer.context.app = app;
  },
});

const buildServerCommand = ({ app }) => ({
  command: 'server',
  describe: 'run server',
  builder: (yargs) => yargs
    .default('h', '127.0.0.1')
    .alias('h', 'host')
    .default('p', Number(process.env.PORT) || 3000)
    .alias('p', 'port'),
  handler: async (argv) => {
    app.listen(argv.port, argv.host, (err, address) => {
      if (err) {
        console.error(err);
        process.exit(1);
      }
      console.log(address);
    });
  },
});

const buildRoutesCommand = ({ app, container }) => ({
  command: 'routes',
  describe: 'display routes',
  handler: async () => {
    const print = _.get(container, 'print', console.log);
    const { routes } = app.router;
    if (_.isEmpty(routes)) {
      print(
        'You don\'t have any routes defined!\n\n'
        + 'Please add some routes in config/routes.yml.',
      );
    } else {
      const formattedRoutes = columnify(
        routes,
        {
          columns: ['name', 'method', 'url', 'pipeline'],
          config: {
            name: {
              headingTransform: () => 'Name',
            },
            method: {
              headingTransform: () => 'Verb',
              dataTransform: _.toUpper,
            },
            url: {
              headingTransform: () => 'URI Pattern',
            },
            pipeline: {
              headingTransform: () => 'Pipeline',
            },
          },
        },
      );
      print(formattedRoutes);
    }
  },
});

const buildGeneratorsCommand = ({ app }) => ({
  command: 'generate <type> <name> [params...]',
  builder: (command) => {
    command.positional('type', {
      describe: 'what you need to create [controller model migration]',
    });
    command.positional('name', {
      describe: 'name of entity',
    });
    command.positional('params', {
      describe: 'list of params that you want apply to your generator',
    });
  },
  handler: ({ type, name, params }) => {
    const { handler } = app.generators.find((generator) => generator.type === type);
    handler({
      app,
      type,
      name,
      params,
    });
  },
});

module.exports = {
  buildConsoleCommand, buildServerCommand, buildRoutesCommand, buildGeneratorsCommand,
};
