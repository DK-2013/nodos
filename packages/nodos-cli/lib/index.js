const Generator = require('yeoman-generator');
const mkdirp = require('mkdirp');
const path = require('path');

export default class extends Generator {
  // The name `constructor` is important here
  constructor(args, opts) {
    // Calling the super constructor is important so our generator is correctly set up
    super(args, opts);

    this.argument('appPath', { type: String, required: true });
  }

  default() {
    const { appPath } = this.options;

    mkdirp(appPath);
    this.sourceRoot(path.resolve(__dirname, '../src/templates'));
    this.destinationRoot(this.destinationPath(appPath));
  }

  writing() {
    const fileNames = [
      'babel.config.js',
      '.gitignore',
      'Makefile',
      'package.json',
      'package-lock.json',
      'config/database.yml',
      'config/routes.yml',
      'config/application.js',
      'config/environments/test.js',
      'config/environments/production.js',
      'config/environments/development.js',
      'bin/test',
      'bin/nodos',
      'app/entities/User.js',
      'app/controllers/users.js',
      'app/middlewares/setLocale.js',
      'app/templates/layouts/application.pug.html',
      'app/templates/users/index.marko',
      'app/templates/users/show.marko',
      'tests/controllers/users.test.js',
    ];

    fileNames.forEach((fileName) => {
      this.fs.copy(
        this.templatePath(fileName),
        this.destinationPath(fileName),
      );
    });
  }
}
