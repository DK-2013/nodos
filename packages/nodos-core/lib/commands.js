const _ = require('lodash');
// const jest = require('jest');
const repl = require('repl');
const columnify = require('columnify');
const log = require('./logger');

// const testCommand = ({ container }) => ({
//   command: 'test [file]',
//   describe: 'run tests',
//   builder: {},
//   handler: (argv) => {
//     // FIXME: not working
//     process.env.NODOS_ENV = 'test';
//     // const rest = argv._.slice(1);
//     const options = [];
//     if (argv.file) {
//       options.push('-f', argv.file);
//     } else {
//       options.push('--testPathPattern', '/tests/');
//     }
//     const jestItem = _.get(container, 'jest', jest);
//     log(options);
//     jestItem.run(options);
//   },
// });

const consoleCommand = ({ app, container }) => ({
  command: 'console',
  describe: 'run console',
  builder: {},
  handler: async () => {
    const replItem = _.get(container, 'repl', repl);
    const replServer = replItem.start({
      prompt: '> ',
    });
    replServer.context.app = app;
  },
});

const serverCommand = ({ app }) => ({
  command: 'server',
  describe: 'run server',
  builder: yargs => yargs
    .default('h', '127.0.0.1')
    .alias('h', 'host')
    .default('p', 3000)
    .alias('p', 'port'),
  handler: async (argv) => {
    app.listen(argv.port, argv.host, (err, address) => {
      if (err) {
        console.error(err);
        process.exit(1);
      }
      console.log(address);
    });
  },
});

const routesCommand = ({ app, container }) => ({
  command: 'routes',
  describe: 'display routes',
  handler: async () => {
    const print = _.get(container, 'print', console.log);
    const { routes } = app.router;
    if (_.isEmpty(routes)) {
      print(
        'You don\'t have any routes defined!\n\n'
        + 'Please add some routes in config/routes.yml.',
      );
    } else {
      const formattedRoutes = columnify(
        routes,
        {
          columns: ['name', 'method', 'url', 'pipeline'],
          config: {
            name: {
              headingTransform: () => 'Name',
            },
            method: {
              headingTransform: () => 'Verb',
              dataTransform: _.toUpper,
            },
            url: {
              headingTransform: () => 'URI Pattern',
            },
            pipeline: {
              headingTransform: () => 'Pipeline',
            },
          },
        },
      );
      print(formattedRoutes);
    }
  },
});

module.exports = { consoleCommand, serverCommand, routesCommand }
