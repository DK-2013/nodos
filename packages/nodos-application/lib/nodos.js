import _ from 'lodash';
import path from 'path';
import * as localCommands from './commands';
import log from './logger';

export default async (projectRoot, env = process.env.NODOS_ENV || 'development') => {
  const appPath = path.join(projectRoot, 'config', 'environments', `${_.capitalize(env)}.js`);
  log(appPath);
  const appModule = await import(appPath);
  log(appModule);
  const app = new appModule.default(projectRoot, env);
  Object.values(localCommands).forEach(app.addCommand);
  return app;
};
