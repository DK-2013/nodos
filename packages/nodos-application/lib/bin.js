import _ from 'lodash';
import yargs from 'yargs';
import log from './logger';
import nodos from './nodos';

export default async (projectRoot, args = process.argv.slice(2), options = {}) => {
  const {
    // exitProcess = true,
    container = {},
    // done = _.noop, // FIXME: hack for testing purposes
  } = options;
  const parser = yargs(args);
  parser.exitProcess(false);
  // parser.fail(console.log);
  parser.demandCommand();
  parser.recommendCommands();
  parser.strict();
  parser.showHelpOnFail(true);
  const nodosItem = _.get(container, 'nodos', nodos);
  const app = await nodosItem(projectRoot);
  await app.start();
  // .epilog(help.trim())
  // .example(example.trim())
  // console.log(app);
  // console.log(app.commandBuilders);
  const commands = app.commands.map(build => build({ app, container }));
  log(commands);
  commands.forEach(c => parser.command(c));
  await parser.argv;
  return app;
};
