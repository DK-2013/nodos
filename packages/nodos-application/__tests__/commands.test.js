import path from 'path';
import { bin } from '../lib';
import * as commands from '../lib/commands';

const projectRoot = path.join(__dirname, '__fixtures__/app');

test('nodos/bin/console', async () => {
  const replServer = { context: {} };
  const container = {
    repl: {
      start: () => replServer,
    },
  };
  const app = await bin(
    projectRoot,
    ['console'],
    {
      container,
    },
  );

  // expect(replServer.context).toHaveProperty('app');
});

test('nodos/bin/test', async () => {
  const run = jest.fn();
  const container = {
    jest: { run },
  };
  const app = await bin(
    projectRoot,
    ['test'],
    {
      container,
    },
  );

  // expect(replServer.context).toHaveProperty('app');
});

test('nodos/bin/server', async () => {
  const fillResult = (port, address, cb) => {
    cb();
  };
  const container = {
    nodos: () => ({
      start: jest.fn(),
      stop: jest.fn(),
      listen: fillResult,
      commands: [commands.serverCommand],
    }),
  };
  const app = await bin(
    projectRoot,
    ['server'],
    {
      container,
    },
  );
});

describe('nodos/bin/routes', () => {
  test('return valid routes presentation', async () => {
    const container = {
      print: (output) => { expect(output).toMatchSnapshot(); },
    };
    const app = await bin(
      projectRoot,
      ['routes'],
      {
        container,
      },
    );
  });

  test('return valid presentation when no routes defined', async () => {
    const container = {
      nodos: () => Promise.resolve({
        start: jest.fn(),
        stop: jest.fn(),
        router: { routes: [] },
        commands: [commands.routesCommand],
      }),
      print: (output) => { expect(output).toMatchSnapshot(); },
    };
    const app = await bin(
      projectRoot,
      ['routes'],
      {
        container,
      },
    );
  });
});
