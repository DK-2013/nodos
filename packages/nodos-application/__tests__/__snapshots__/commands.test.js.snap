// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nodos/bin/routes return valid presentation when no routes defined 1`] = `
"You don't have any routes defined!

Please add some routes in config/routes.yml."
`;

exports[`nodos/bin/routes return valid routes presentation 1`] = `
"Name                Verb   URI Pattern                             Pipeline
apiUsers            GET    /api/users                              browser 
buildApiUser        GET    /api/users/build                        browser 
apiUsers            POST   /api/users                              browser 
apiUser             GET    /api/users/:id                          browser 
editApiUser         GET    /api/users/:id/edit                     browser 
apiUser             PATCH  /api/users/:id                          browser 
apiUser             PUT    /api/users/:id                          browser 
apiUser             DELETE /api/users/:id                          browser 
users               GET    /users                                  browser 
buildUser           GET    /users/build                            browser 
users               POST   /users                                  browser 
user                GET    /users/:id                              browser 
editUser            GET    /users/:id/edit                         browser 
user                PATCH  /users/:id                              browser 
user                PUT    /users/:id                              browser 
user                DELETE /users/:id                              browser 
posts               GET    /posts                                  browser 
buildPost           GET    /posts/build                            browser 
posts               POST   /posts                                  browser 
post                GET    /posts/:id                              browser 
editPost            GET    /posts/:id/edit                         browser 
post                PATCH  /posts/:id                              browser 
post                PUT    /posts/:id                              browser 
post                DELETE /posts/:id                              browser 
buildSession        GET    /session/build                          browser 
sessions            POST   /session                                browser 
session             GET    /session                                browser 
editSession         GET    /session/edit                           browser 
session             PATCH  /session                                browser 
session             PUT    /session                                browser 
session             DELETE /session                                browser 
articles            GET    /articles                               browser 
buildArticle        GET    /articles/build                         browser 
articles            POST   /articles                               browser 
article             GET    /articles/:id                           browser 
editArticle         GET    /articles/:id/edit                      browser 
article             PATCH  /articles/:id                           browser 
article             PUT    /articles/:id                           browser 
article             DELETE /articles/:id                           browser 
articleComments     GET    /articles/:article_id/comments          browser 
buildArticleComment GET    /articles/:article_id/comments/build    browser 
articleComments     POST   /articles/:article_id/comments          browser 
articleComment      GET    /articles/:article_id/comments/:id      browser 
editArticleComment  GET    /articles/:article_id/comments/:id/edit browser 
articleComment      PATCH  /articles/:article_id/comments/:id      browser 
articleComment      PUT    /articles/:article_id/comments/:id      browser 
articleComment      DELETE /articles/:article_id/comments/:id      browser "
`;
